e2e/test/iothub/service/ConfigurationsClientE2ETests.cs

```
@@ -86,5 +88,162 @@ public async Task ConfigurationOperations_Work()
+                        Queries = { { "successfullyConfigured", "select deviceId from devices where properties.reported.x = 4" } }
+                    },
+                };
+
+                configuration = await serviceClient.Configurations.CreateAsync(configuration).ConfigureAwait(false);
+                configCreated = true;
+                ETag oldEtag = configuration.ETag;
+
+                configuration.Priority = 3;
+
+                configuration = await serviceClient.Configurations.SetAsync(configuration).ConfigureAwait(false);
+
+                configuration.ETag = oldEtag;
+
+                try
+                {
+                    // set the 'onlyIfUnchanged' flag to true to check that, with an out of date ETag, the request throws a PreconditionFailedException.
+                    await serviceClient.Configurations.DeleteAsync(configuration, true).ConfigureAwait(false);
+                    throw new AssertFailedException("Expected test to throw a precondition failed exception since it updated a configuration with an out of date ETag");
+                }
+                catch (DeviceMessageLockLostException)
+                {
+                    // expected thrown exception, continue the test without throwing anything
+                }
+
+                try
+                {
+                    // set the 'onlyIfUnchanged' flag to false to check that, even with an out of date ETag, the request performs without exception.
+                    await serviceClient.Configurations.DeleteAsync(configuration, false).ConfigureAwait(false);
+                }
+                catch (DeviceMessageLockLostException)
+                {
+                    throw new AssertFailedException("Did not expect test to throw a precondition failed exception since 'onlyIfUnchanged' was set to false");
+                }
+            }
+            finally
+            {
+                try
+                {
+                    if (configCreated)
```

No need to check this. Just attempt the delete operation and let it throw a not found exception that the test ignores